src/tess.py

```diff
@@ -0,0 +1,364 @@
+            candidate += 1
+        
+        # if there are multiple likely candidates, throw warning
+        close_mags = 0
+        for i in range(len(catalog_data)):
+            gm = catalog_data['GAIAmag'][i]
+            dst = catalog_data['dstArcSec'][i]
+            if abs(gm - mag) < MAGDIFF_THRESHOLD and dst < DST_THRESHOLD:
+                close_mags += 1
+        if close_mags > 1:
+            warnings.warn(f"Multiple ({close_mags}) sources with similar " +
+                            "magnitudes in cone search." +
+                            " Might select incorrect source.")
+
+        TESS_ID = catalog_data['ID'][candidate]
+        kwargs = {
+            'objectname': 'TIC ' + TESS_ID,
+            'radius': 0.1,
+            'obs_collection': 'TESS',
+            # 'dataproduct_type': 'timeseries'
+        }
+        data_query = self._try_query(Observations.query_criteria, kwargs)
+        if len(catalog_data) == 0:
+            print("No data found for object " + 'TIC ' + TESS_ID + '.')
+            return pd.DataFrame()
+        if 'timeseries' not in data_query['dataproduct_type']:
+            print("No timeseries data found for object " + 
+                    'TIC ' + TESS_ID + '.')
+            return pd.DataFrame()
+        
+        lc_indices = []
+        for i, uri in enumerate(data_query['dataURL']):
+            if isinstance(uri, str) and uri[-5:-7] == 'lc':
+                lc_indices.append(i)
+        print(f"Found {len(lc_indices)} light curve(s) for this source.")
+
+        base_url = "https://mast.stsci.edu/api/v0.1/Download/file?uri="
+
+        # how to properly aggregate data if more than 1 LC?
+        # do we just add rows naively?
+        df_list = []
+        for i in lc_indices:
+            uri = data_query['dataURL'][i]
+            table = Table.read(base_url + uri, format='fits')
+            df_list.append(table.to_pandas())
+        data = pd.concat(df_list, ignore_index=True)
+
+        first_uri = data_query['dataURL'][lc_indices[0]]
+        # what other metadata should we save from a FITS file?
+        altdata = {}
+        with fits.open(base_url + first_uri, cache=False) as hdul:
+            altdata['lightcurve columns'] = hdul[1].data.columns.__str__()
+            altdata['aperture matrix'] = hdul[2].data
+        
+        return data, altdata
+
+
+    def _try_query(self, query_fn, kwargs):
+        """
+        Makes an astroquery request repeatedly, ignoring any timeout errors.
+        Returns first successful response, otherwise raises TimeoutError.
+        """
+        # maybe try using multiprocessing to terminate after 10 secs?
+        tries = 1
+        while tries <= 10:
+            try:
+                print(f"Making query request, attempt {tries}/10 ...")
+                ret = query_fn(**kwargs)
+                return ret
+            except TimeoutError as e:
+                # print(e)
+                print(f"Request timed out.")
```

Comment:
The print statements should definitely be in verbose or possibly `verbose>2` or something like that.